main.tf

```diff
@@ -228,7 +233,7 @@ locals {
 
 resource "ibm_is_public_gateway" "gateway" {
   for_each       = local.gateway_object
-  name           = var.prefix != null ? "${var.prefix}-${var.name}-public-gateway-${each.key}" : "${var.name}-public-gateway-${each.key}"
+  name           = var.prefix != null ? "${var.prefix}-${var.name}-public-gateway-${each.key}" : "${var.public_gateway_name}-${each.key}"
```

Comment:
I don't see any reason to add `var.public_gateway_name` - you can achieve this naming with the current code using `var.name`. There can be multiple public gateways here, so we will always be appending `${each.value.name}` to the name here

Reply:
same as https://github.com/terraform-ibm-modules/terraform-ibm-landing-zone-vpc/pull/685#discussion_r1426907203